src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -381,18 +381,36 @@ object RuleUtils {
       filesAppended: Seq[Path]): LogicalPlan = {
     // Transform the location of LogicalRelation with appended files.
     val planForAppended = originalPlan transformDown {
-      case baseRelation @ LogicalRelation(fsRelation: HadoopFsRelation, baseOutput, _, _) =>
+      case baseRelation @ LogicalRelation(
+            fsRelation @ HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, _, _),
+            baseOutput,
+            _,
+            _) =>
+        val options = if (location.partitionSchema.isEmpty) {
+          Map[String, String]()
+        } else {
+          // Retrieve basePath from a partition path.
+          val basePath = location.partitionSpec.partitionColumns
+            .foldLeft(location.partitionSpec.partitions.head.path) { (path, _) =>
+              path.getParent
```

Comment:
Since we are more or less reverse engineering to compute base path, can we factor out to a function and add more descriptive comments? For example:
```scala
  private def extractBasePath(partitionSpec: PartitionSpec): Option[String] = {
    if (partitionSpec == PartitionSpec.emptySpec) {
      None
    } else {
      // For example, we could have the following in PartitionSpec:
      //   - partition columns = "col1", "col2"
      //   - partitions: "/path/col1=1/col2=1", "/path/col1=1/col2=2", etc.
      // , and going up the same number of directory levels as the number of partition columns
      // will compute the base path. Note that PartitionSpec.partitions will always contain
      // all the partitions in the path, so "partitions.head" is taken as an initial value.
      val basePath = partitionSpec.partitionColumns
        .foldLeft(partitionSpec.partitions.head.path)((path, _) => path.getParent)
      Some(basePath.toString)
    }
  }
```

Reply:
Question: do we need to worry about the case where there are multiple base paths in `PartitioningAwareFileIndex.basePaths`?

Reply:
Seems we don't: https://github.com/apache/spark/blob/d6a68e0b67ff7de58073c176dd097070e88ac831/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala#L166